clang/docs/ReleaseNotes.rst

```diff
@@ -583,6 +583,8 @@ Improvements to Clang's diagnostics
 - For an rvalue reference bound to a temporary struct with an integer member, Clang will detect constant integer overflow
   in the initializer for the integer member (#GH46755).
 
+- Clang now prevents errors for deduction guides with deduced type aliases (#GH54909).
```

Comment:
This isn't clear enough to me: Someone might ask why we should 'prevent' errors in such cases, and what the rationale is behind it. It might help to reference some standardese to make it unimpeachable.

Reply:
@zyn0217 thanks for the feedback. I've added the requested changes.